packages/browser/src/index.ts

```diff
@@ -29,17 +33,54 @@ export type AccessToken = {
   expiresAt: number; // Unix Timestamp in seconds
 };
 
+export const LogtoSignInSessionItemSchema = type({
+  redirectUri: string(),
+  codeVerifier: string(),
+  state: string(),
+});
+
+export type LogtoSignInSessionItem = Infer<typeof LogtoSignInSessionItemSchema>;
+
 export default class LogtoClient {
   protected accessTokenMap = new Map<string, AccessToken>();
   protected refreshToken?: string;
   protected idToken?: string;
   protected logtoConfig: LogtoConfig;
   protected oidcConfig?: OidcConfigResponse;
 
+  private readonly _signInSession: Optional<LogtoSignInSessionItem>;
+
   constructor(logtoConfig: LogtoConfig) {
     this.logtoConfig = logtoConfig;
   }
 
+  protected get signInSession(): LogtoSignInSessionItem {
+    const logtoKey = getLogtoKey(this.logtoConfig.clientId);
```

Comment:
Remember we need do it in init process for refresh / id token due to performance concern. I'm not sure if we need align this behavior over session or keep as it is. @logto-io/eng any idea?

Reply:
Look like this? PTAL

<img width="539" alt="image" src="https://user-images.githubusercontent.com/10594507/153992433-0530e60a-0d13-4d7a-bdf9-0338eaf41a97.png">

Reply:
Summary on the Notion [documentation](https://www.notion.so/silverhand/TypeScript-Convention-5a8f2a6da25140318e0adbe74e6756bc#f19841d774294b0583fe1a1de1968922)